lib/dor/text_extraction/speech_to_text.rb

```diff
@@ -124,10 +137,11 @@ def output_location
         "#{job_id}/output"
       end
 
-      # given a filename, look in the list of files that can be sent for speech to text, examine the cocina structural
+      # given a s3 filename, look in the list of files that can be sent for speech to text, examine the cocina structural
       #  and return the languageTag for the file (or nil if no language is set)
-      def language_tag(filename)
-        stt_files.find { |file| file.filename == filename }&.languageTag
+      #  we need to account for the way we rename files by adding the extension when sending to S3
+      def language_tag(s3_filename)
+        stt_files.find { |file| file.filename == cocina_filename(s3_filename) }&.languageTag
```

Comment:
use our new method so that when we have an s3 filename, we can find the matching one in cocina

Reply:
Ah, so this is why the language tag wasn't being found -- because it was looking it up using the original filename instead of the one for s3?

Reply:
yes, exactly -- and it wasn't erroring, it just wasn't finding the language tag and so was never adding it.